src/components/rangeslider/draw.js

```diff
@@ -0,0 +1,424 @@
+                    pixelMin = offsetX;
+                    pixelMax = offsetX + delta;
+                    break;
+            }
+
+            if(pixelMax < pixelMin) {
+                var tmp = pixelMax;
+                pixelMax = pixelMin;
+                pixelMin = tmp;
+            }
+
+            opts._pixelMin = pixelMin;
+            opts._pixelMax = pixelMax;
+
+            setCursor(d3.select(dragCover), cursor);
+            setDataRange(rangeSlider, gd, axisOpts, opts);
+        }
+
+        function mouseUp() {
+            dragCover.removeEventListener('mousemove', mouseMove);
+            dragCover.removeEventListener('mouseup', mouseUp);
+            Lib.removeElement(dragCover);
+        }
+    });
+}
+
+function setDataRange(rangeSlider, gd, axisOpts, opts) {
+
+    function clamp(v) {
+        return Lib.constrain(v, opts.range[0], opts.range[1]);
+    }
+
+    var dataMin = clamp(opts.p2d(opts._pixelMin)),
+        dataMax = clamp(opts.p2d(opts._pixelMax));
+
+    window.requestAnimationFrame(function() {
+        Plotly.relayout(gd, 'xaxis.range', [dataMin, dataMax]);
+    });
+}
+
+function setPixelRange(rangeSlider, gd, axisOpts, opts) {
+
+    function clamp(v) {
+        return Lib.constrain(v, 0, opts._width);
+    }
+
+    var pixelMin = clamp(opts.d2p(axisOpts.range[0])),
+        pixelMax = clamp(opts.d2p(axisOpts.range[1]));
+
+    rangeSlider.select('rect.' + constants.slideBoxClassName)
+        .attr('x', pixelMin)
+        .attr('width', pixelMax - pixelMin);
+
+    rangeSlider.select('rect.' + constants.maskMinClassName)
+        .attr('width', pixelMin);
+
+    rangeSlider.select('rect.' + constants.maskMaxClassName)
+        .attr('x', pixelMax)
+        .attr('width', opts._width - pixelMax);
+
+    rangeSlider.select('g.' + constants.grabberMinClassName)
+        .attr('transform', 'translate(' + (pixelMin - constants.handleWidth - 1) + ',0)');
+
+    rangeSlider.select('g.' + constants.grabberMaxClassName)
+        .attr('transform', 'translate(' + pixelMax + ',0)');
+}
+
+function drawBg(rangeSlider, gd, axisOpts, opts) {
+    var bg = rangeSlider.selectAll('rect.' + constants.bgClassName)
+        .data([0]);
+
+    bg.enter().append('rect')
+        .classed(constants.bgClassName, true)
+        .attr({
+            x: 0,
+            y: 0,
+            'shape-rendering': 'crispEdges'
+        });
+
+    var borderCorrect = (opts.borderwidth % 2) === 0 ?
+            opts.borderwidth :
+            opts.borderwidth - 1;
+
+    var offsetShift = -opts._offsetShift;
+
+    bg.attr({
+        width: opts._width + borderCorrect,
+        height: opts._height + borderCorrect,
+        transform: 'translate(' + offsetShift + ',' + offsetShift + ')',
+        fill: opts.bgcolor,
+        stroke: opts.bordercolor,
+        'stroke-width': opts.borderwidth,
+    });
+}
+
+function drawRangePlot(rangeSlider, gd, axisOpts, opts) {
+    var rangePlots = rangePlot(gd, opts._width, opts._height);
```

Comment:
Commits in https://github.com/plotly/plotly.js/compare/range-slider-range-plot indeed make the current `rangePlot()` obsolete. Those commits deserve a PR of their own - coming after this PR is merged.